File: python/src/workshop_api/fitness/managing_plans/shared/plan_response.py
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from workshop_api.fitness.managing_plans.shared.plan_entity import PlanOrmModel


class PlanResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: UUID
    title: str
    description: str
    duration_in_months: int = Field(
        alias="durationInMonths",
        serialization_alias="durationInMonths",
    )
    price: Decimal

    @staticmethod
    def from_entity(entity: PlanOrmModel) -> PlanResponse:
        return PlanResponse(
            id=entity.id,
            title=entity.title,
            description=entity.description,
            durationInMonths=entity.duration_in_months,
            price=entity.price,
        )

File: java/src/main/java/com/workshop/architecture/fitness/managing_plans/PlanResponse.java
package com.workshop.architecture.fitness.managing_plans;

import java.math.BigDecimal;
import java.util.UUID;

public record PlanResponse(
        UUID id,
        String title,
        String description,
        int durationInMonths,
        BigDecimal price
) {
    public static PlanResponse fromEntity(PlanEntity entity) {
        return new PlanResponse(
                entity.getId(),
                entity.getTitle(),
                entity.getDescription(),
                entity.getDurationInMonths(),
                entity.getPrice()
        );
    }
}
